rust/automerge/src/query/list_state.rs

```diff
@@ -1,6 +1,43 @@
+use crate::marks::MarkData;
 use crate::op_tree::{LastInsert, OpTreeNode};
 use crate::query::QueryResult;
-use crate::types::{Key, ListEncoding, Op};
+use crate::types::{Key, ListEncoding, Op, OpId, OpType};
+use fxhash::FxBuildHasher;
+use std::collections::HashMap;
+use std::ops::{Deref, DerefMut};
+
+#[derive(Debug, Clone, Default, PartialEq)]
+pub(crate) struct MarkMap<'a> {
+    map: HashMap<OpId, &'a MarkData, FxBuildHasher>,
+}
+
+impl<'a> Deref for MarkMap<'a> {
```

Comment:
Implementing `DeRef` for things which aren't smart pointers is generally a bad idea. Per [the docs](https://doc.rust-lang.org/std/ops/trait.Deref.html) on `Deref`:

> Deref should only be implemented for smart pointers to avoid confusion

Can we just wrap the relevant methodos from `HashMap`?